Automattic/wp-calypso
# Site Migration: Remove deprecated instructions step

## Proposed Changes
* Remove the deprecated site migration instructions
* Move the shared components to the i2 folder.

NOTE: In this PR I am not renaming the step slug to avoid any issues. We can do it in a separate PR. 

## Why are these changes being made?
* Code-based maintenance, since the removed step is not used and should not be used by any new initiative. 

## Testing Instructions
* Run the site migration instructions and check if the page is still working.  There are automated tests covering it.

## Pre-merge Checklist
